geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionAdvisor.java

```diff
@@ -836,22 +846,20 @@ public void waitForCurrentOperations(Logger alertLogger, long warnMS, long sever
       long now = System.currentTimeMillis();
       if ((!warned) && System.currentTimeMillis() >= warnTime) {
         warned = true;
-        alertLogger.warn("This operation has been stalled for {} milliseconds waiting for "
+        alertLogger.warn("This thread has been stalled for {} milliseconds waiting for "
             + "current operations to complete.", warnMS);
+        logger.debug("Waiting for these threads: {}", previousVersionOperationThreads);
+        logger.debug("New version threads are {}", currentVersionOperationThreads);
       } else if (warned && !severeAlertIssued && (now >= severeAlertTime)) {
         // OSProcess.printStacks(0);
-        alertLogger.fatal("This operation has been stalled for {} milliseconds "
+        alertLogger.fatal("This thread has been stalled for {} milliseconds "
             + "waiting for current operations to complete.  Something may be blocking operations.",
             severeAlertMS);
+        logger.debug("Waiting for these threads: {}", previousVersionOperationThreads);
```

Comment:
All other uses of these new hash maps are protected by synchronizing on opCountLock. Should that happen here also?
if (logger.debugEnabled()) {
  synchronize (this.opCountLock) {
    logger.debug("Waiting for these threads: {}", previousVersionOperationThreads);
    ...
  }
}